### PR TITLE
[Feature] Adapting the catalog in fe

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #
related with https://github.com/StarRocks/starrocks/issues/6514

related with https://github.com/StarRocks/starrocks/issues/5062

## Problem Summary(Required) ：

DLA New Framework add catalog level. We need to adapt the catalog concept where we use the hive connector table at present. we only support querying hive connector table in this milestone. Therefore, we only adapt the catalog in the new parser and new analyzer. we also need refresh_stmt in old parser when we using hive connector table. we will support it in next pr. ref https://github.com/StarRocks/starrocks/issues/6509.
And other stmt like describe table on external catalog https://github.com/StarRocks/starrocks/issues/6510. we will support it  step by step in future.

The main point of adaption:
- Add catalog field in TableName
- Use `MetadataMgr.getDb` or `MetadataMgr.getTable` when getting metadata.
- Remove cluster_name from dbname when process table in external catalog.
- Permission related will be merged after this pr https://github.com/StarRocks/starrocks/pull/6496

### CODE REVIEW

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by caneGuy** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
cc @HangyuanLiu

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java [hunk 1]
```diff
@@ -339,7 +340,7 @@ public Database getDb(String dbName) {
     private Database loadDatabase(String dbName) throws TException {
         // Check whether the db exists, if not, an exception will be thrown here
         org.apache.hadoop.hive.metastore.api.Database db = client.getDb(dbName);
-        if (db.getName() == null) {
+        if (db == null || db.getName() == null) {
```
Comment (RowenWoo) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:343]:
```suggestion
        if (db == null && db.getName() == null) {
```

**Review by RowenWoo — CHANGES_REQUESTED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -2353,10 +2353,13 @@ private QualifiedName getQualifiedName(StarRocksParser.QualifiedNameContext cont
     }
 
     private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
-        if (qualifiedName.getParts().size() == 2) {
-            return new TableName(qualifiedName.getParts().get(0), qualifiedName.getParts().get(1));
-        } else if (qualifiedName.getParts().size() == 1) {
-            return new TableName(null, qualifiedName.getParts().get(0));
+        List<String> parts = qualifiedName.getParts();
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2356]: nit: plz add some comments

File: fe/fe-core/src/main/java/com/starrocks/sql/common/MetaUtils.java [hunk 3]
```diff
@@ -40,7 +47,9 @@ public static void normalizationTableName(ConnectContext connectContext, TableNa
             if (Strings.isNullOrEmpty(connectContext.getClusterName())) {
                 throw new SemanticException("No cluster name");
             }
-            tableName.setDb(ClusterNamespace.getFullName(connectContext.getClusterName(), tableName.getDb()));
+            if (CatalogMgr.isInternalCatalog(tableName.getCatalog())) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/MetaUtils.java:50]: nit: extract a function, this statement is used everywhere

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMgr.java [hunk 4]
```diff
@@ -10,12 +10,15 @@
...
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 // ConnectorMgr is responsible for managing all ConnectorFactory, and for creating Connector
 public class ConnectorMgr {
     private static final Logger LOG = LogManager.getLogger(MetadataMgr.class);
     private final ConcurrentHashMap<String, Connector> connectors = new ConcurrentHashMap<>();
     private final ConcurrentHashMap<String, ConnectorFactory> connectorFactories = new ConcurrentHashMap<>();
+    private final ReadWriteLock connectorLock = new ReentrantReadWriteLock();
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMgr.java:21]: nit: use db_lock instead?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2356]: done

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 120 / 133 (90.23%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|21|26|80.77%|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|24|27|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|9|10|90.00%|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|16|17|94.12%|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/analysis/Analyzer.java|1|1|100.00%|

**Review by DorianZheng — APPROVED** [review]
LGTM

**Review by caneGuy — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**